src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -679,10 +679,9 @@ protected Task FireOnStarting()
             lock (_onStartingSync)
             {
                 onStarting = _onStarting;
-                _onStarting = null;
```

Comment:
_onStartingSync and _onCompletedSync are now pretty worthless after this change. Previously, unsetting _onStarting inside the lock would prevent race conditions from occurring if OnStarting/Completed was called while FireOnStarting/Completed was iterating over the stack.

It can be argued that OnStarting/Completed should only be called from the request thread, so I'm OK with removing the locks.